### ISSUE TITLE
ts: embedded models and collections are generating incorrectly

### ISSUE THREAD

**Issue by mickeyreiss** [mickeyreiss/firemodel#9]
1. embedded models should be the interface directly, not a map from strings to the interface
2. collections should generate a collection reference

**Comment by ZakTax** [comment]
schema:
```
  ClientProfile clientProfile;
```
generates
```
    /** TODO: Add documentation to client_profile. */
    clientProfile: { [key: string]: IClientProfile; };
```
should generate
```
    /** TODO: Add documentation to client_profile. */
    clientProfile: IClientProfile;
```
schema
```
  Proposal proposals;
```
should be
```
  collection<Proposal> proposals;
```
generates
```
    /** TODO: Add documentation to projects. */
    projects: { [key: string]: IProject; };
```
should generate
```
    /** TODO: Add documentation to projects. */
    projects: CollectionReference<IProduct>;
```

**Comment by mickeyreiss** [comment]
Resolved in 3921c10c0f9a49b61950c23c960ddcce96a6626a and 553fde81e626df0085578e4821b08622bb01a24a
